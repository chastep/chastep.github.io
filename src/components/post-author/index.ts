export { PostAuthor } from "./post-author";
