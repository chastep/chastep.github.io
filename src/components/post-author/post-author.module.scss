@use "src/assets/styles/mixins";
@use "src/assets/styles/variables";

.post-author {
  border-top: 1px solid hsl(var(--color-border));
  max-width: variables.$layout-post-width;
  @include mixins.padding-top(1);
  @include mixins.line-height(1);
  @include mixins.margin-top(1);
  @include mixins.margin-bottom(2);

  .description {
    .x {
      display: block;
      text-decoration: underline;
    }
  }
}

@include mixins.breakpoint-sm {
  .post-author {
    margin-left: auto;
    margin-right: auto;
  }
}
