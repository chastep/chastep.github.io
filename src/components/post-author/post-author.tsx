import React, { useMemo } from "react";

import { getContactHref } from "@/utils/get-contact-href";
import { useSiteMetadata } from "@/hooks/use-site-metadata";

import * as styles from "./post-author.module.scss";

const PostAuthor = () => {
  const { author } = useSiteMetadata();
  const x = useMemo(
    () => author.contacts.find(({ name }) => name === "x"),
    [author],
  );

  return (
    <div className={styles.postAuthor}>
      <p className={styles.description}>
        {author.description}
        {x ? (
          <a
            className={styles.x}
            href={getContactHref(x.name, x.contact)}
            rel="noopener noreferrer"
            target="_blank"
          >
            <strong>{author.title}</strong> on X
          </a>
        ) : null}
      </p>
    </div>
  );
};

export { PostAuthor };
