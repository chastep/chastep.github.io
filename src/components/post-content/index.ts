export { PostContent } from "./post-content";
