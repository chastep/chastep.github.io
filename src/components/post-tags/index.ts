export { PostTags } from "./post-tags";