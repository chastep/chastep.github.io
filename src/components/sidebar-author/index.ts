export { SidebarAuthor } from "./sidebar-author";