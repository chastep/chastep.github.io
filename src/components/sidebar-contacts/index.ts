export { SidebarContacts } from "./sidebar-contacts";
