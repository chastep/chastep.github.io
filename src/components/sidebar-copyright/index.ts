export { SidebarCopyright } from "./sidebar-copyright";
