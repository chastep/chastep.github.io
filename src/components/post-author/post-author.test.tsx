import React from "react";
import { StaticQuery, useStaticQuery } from "gatsby";
import { test, describe, expect, beforeEach, mock } from "bun:test";

import * as mocks from "@/mocks";
import { PostAuthor } from "@/components/post-author";
import { createSnapshotsRenderer } from "@/utils/render-with-coil-provider";

const mockedUseStaticQuery = useStaticQuery as ReturnType<typeof mock>;
const mockedStaticQuery = StaticQuery as unknown as ReturnType<typeof mock>;

describe("PostAuthor", () => {
  beforeEach(() => {
    mockedStaticQuery.mockImplementationOnce(({ render }) =>
      render(mocks.siteMetadata),
    );

    mockedUseStaticQuery.mockReturnValue(mocks.siteMetadata);
  });

  test("renders correctly", () => {
    const tree = createSnapshotsRenderer(<PostAuthor />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
