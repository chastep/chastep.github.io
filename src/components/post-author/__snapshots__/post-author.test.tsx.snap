// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`PostAuthor renders correctly 1`] = `
{
  "$$typeof": Symbol(react.test.json),
  "children": [
    {
      "$$typeof": Symbol(react.test.json),
      "children": [
        "Pellentesque odio nisi, euismod in, pharetra a, ultricies in, diam. Sed arcu.",
        {
          "$$typeof": Symbol(react.test.json),
          "children": [
            {
              "$$typeof": Symbol(react.test.json),
              "children": [
                "John Doe",
              ],
              "props": {},
              "type": "strong",
            },
            " on X",
          ],
          "props": {
            "className": undefined,
            "href": "https://www.x.com/#",
            "rel": "noopener noreferrer",
            "target": "_blank",
          },
          "type": "a",
        },
      ],
      "props": {
        "className": undefined,
      },
      "type": "p",
    },
  ],
  "props": {
    "className": undefined,
  },
  "type": "div",
}
`;
